src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -318,7 +318,10 @@ def recordInCouch(self, jobs, newstate, oldstate, updatesummary = False):
                     jobState = "jobfailed"
                 else:
                     jobState = newstate
-                    
+                
+                # there is race condition updating couch record location and job is completed0
+                # for the fast fail job, it could miss the location update
+                job["location"] = job["fwjr"].getSiteName() or job.get("location", "UnKnown")     
```

Comment:
So, if there is a site in the fwjr, it gets assigned. Otherwise it gets the original location for the job. Is that right? Can this job have multiple locations?

I think it's better to avoid these capital letters, so `UnKnown` to `Unknown`. There is a typo in the `completed0` comment as well